### PR TITLE
fix: check if there is data

### LINKED ISSUE #21: [FEATURE] Display the last fetched data for daily
### Description

Data is collect at 8am UTC to make sure we have today's new data, as we don't know when GitHub releases the new trending data.

If visiting the root of the project `/` before 8am UTC, there is no data for today, we should display the latest data.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Bashamega**
closes #21 
Can you add the label `hacktoberfest-accepted`?

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Sure, label added

File: src/app.service.ts [hunk 1]
```diff
@@ -12,10 +12,25 @@ export class AppService {
...
 
   async getStats(): Promise<any> {
-    return this.prisma.trending.groupBy({
+    const stats = await this.prisma.trending.groupBy({
       by: ['createdAt', 'type'],
       _count: { id: true },
     });
+    if(stats.length > 0){
+      return stats
+    }else{
```
Comment (eddiejaoude) [inline src/app.service.ts]: I think the `}else{` is not needed because the `if` has a `return`

File: src/app.service.ts [hunk 2]
```diff
@@ -12,10 +12,25 @@ export class AppService {
...
+      const getYesterdaysDate = () => {
```
Comment (eddiejaoude) [inline src/app.service.ts]: Does this need to be a function?

File: src/app.service.ts (hunk 1 above)
Comment (Bashamega) [reply src/app.service.ts]: Your are correct. I will fix it.

File: src/app.service.ts (hunk 2 above)
Comment (Bashamega) [reply src/app.service.ts]: Not really. I can fix it.

File: src/app.service.ts (hunk 1 above)
Comment (Bashamega) [reply src/app.service.ts]: Done

File: src/app.service.ts (hunk 2 above)
Comment (Bashamega) [reply src/app.service.ts]: Done

Comment (eddiejaoude) [reply src/app.service.ts]: Thank you 👍

File: src/app.service.ts (hunk 1 above)
Comment (eddiejaoude) [reply src/app.service.ts]: Thanks

File: src/app.service.ts [hunk 3]
```diff
@@ -12,10 +12,23 @@ export class AppService {
...
+      return stats;
+    }
+    const yesterday = new Date();
+    yesterday.setDate(yesterday.getDate() - 1);
+
+    const year = yesterday.getFullYear();
+    const month = String(yesterday.getMonth() + 1).padStart(2, '0'); // Months are 0-indexed
+    const day = String(yesterday.getDate()).padStart(2, '0');
+
+    const getYesterdaysDate = `${year}-${month}-${day}`;
```
Comment (eddiejaoude) [inline src/app.service.ts]: I think we can use `format` like we use already in the file...

https://github.com/EddieHubCommunity/github-trending-repos/blob/fb177abd6728740eada3ad30e363ca101ce825d9/src/app.service.ts#L36

Comment (Bashamega) [reply src/app.service.ts]: I have used the format function. I didn't see it. LOL!

Comment (eddiejaoude) [reply src/app.service.ts]: thanks

**Review by eddiejaoude — APPROVED** [review]
Thank you

**Pull request merged**